src/main/ui/chat-ui.gd

```diff
@@ -0,0 +1,264 @@
+"""
+func _show_nametag():
+	$Holder/ChatRect/NametagRect.frame = randi() % 4
+	$Holder/ChatRect/NametagRect.flip_h = randf() > 0.5
+	$Holder/ChatRect/NametagRect.flip_v = randf() > 0.5
+	$Holder/ChatRect/NametagRect.self_modulate = _get_border_color()
+	$Holder/ChatRect/NametagRect/NametagLabelS.set("custom_colors/font_color", Color.black if _is_dark() else Color.white)
+	_nametag_labels[_nametag_size].visible = true
+	$Holder/ChatRect/NametagRect.texture = _nametag_rect_textures[_nametag_size]
+	
+	var nametag_right: bool = _accent_def["nametag_right"] if _accent_def.has("nametag_right") else false
+	if nametag_right:
+		if _nametag_size == NametagSize.SMALL:
+			$Holder/ChatRect/NametagRect.rotation_degrees = 3
+			match(_chat_size):
+				ChatSize.SMALL: $Holder/ChatRect/NametagRect.position = Vector2(493, -420)
+				ChatSize.MEDIUM: $Holder/ChatRect/NametagRect.position = Vector2(942, -439)
+				ChatSize.LARGE: $Holder/ChatRect/NametagRect.position = Vector2(883, -550)
+				ChatSize.EXTRA_LARGE: $Holder/ChatRect/NametagRect.position = Vector2(1412, -550)
+		else:
+			$Holder/ChatRect/NametagRect.rotation_degrees = 1
+			match(_chat_size):
+				ChatSize.SMALL: $Holder/ChatRect/NametagRect.position = Vector2(126, -431)
+				ChatSize.MEDIUM: $Holder/ChatRect/NametagRect.position = Vector2(558, -448)
+				ChatSize.LARGE: $Holder/ChatRect/NametagRect.position = Vector2(476, -553)
+				ChatSize.EXTRA_LARGE: $Holder/ChatRect/NametagRect.position = Vector2(957, -553)
+	else:
+		if _nametag_size == NametagSize.SMALL:
+			$Holder/ChatRect/NametagRect.rotation_degrees = -3
+			match(_chat_size):
+				ChatSize.SMALL: $Holder/ChatRect/NametagRect.position = Vector2(-497, -410)
+				ChatSize.MEDIUM: $Holder/ChatRect/NametagRect.position = Vector2(-937, -439)
+				ChatSize.LARGE: $Holder/ChatRect/NametagRect.position = Vector2(-947, -550)
+				ChatSize.EXTRA_LARGE: $Holder/ChatRect/NametagRect.position = Vector2(-1437, -530)
+		else:
+			$Holder/ChatRect/NametagRect.rotation_degrees = -1
+			match(_chat_size):
+				ChatSize.SMALL: $Holder/ChatRect/NametagRect.position = Vector2(-151, -423)
+				ChatSize.MEDIUM: $Holder/ChatRect/NametagRect.position = Vector2(-550, -431)
+				ChatSize.LARGE: $Holder/ChatRect/NametagRect.position = Vector2(-550, -561)
+				ChatSize.EXTRA_LARGE: $Holder/ChatRect/NametagRect.position = Vector2(-1072, -562)
+
+
+func _is_dark() -> bool:
+	return _accent_def["dark"] if _accent_def.has("dark") else false
+
+
+func _get_accent_color() -> Color:
+	var accent_color: Color = _accent_def["color"] if _accent_def.has("color") else Color.gray
+	if _is_dark():
+		# accent color is a darker version of the input color
+		accent_color.v = lerp(accent_color.v, 0.33, 0.8)
+	else:
+		# accent color is a lighter, more saturated version of the input color
+		accent_color.v = lerp(accent_color.v, 0.67, 0.8)
+		accent_color.s = pow(accent_color.s, 0.22)
+	return accent_color
+
+
+func _get_border_color() -> Color:
```

Comment:
Could these types of helper methods be moved onto a new Accent object, or something like that?

Reply:
Refactored helper methods onto new `ChatAppearance` class.